fenixedu-ist-giaf-contracts/src/main/java/pt/ist/fenixedu/contracts/domain/accessControl/DepartmentPresidentStrategy.java

```diff
@@ -86,8 +86,29 @@ public static Person getCurrentDepartmentPresident(Department department) {
         return null;
     }
 
+    public static Set<Person> getCurrentDepartmentPresidents(Department department) {
+        Set<Person> departmentPresidents = new HashSet<Person>();
+        for (final Accountability accountability : department.getDepartmentUnit().getChildsSet()) {
+            if (accountability instanceof PersonFunction && accountability.isActive()) {
+                final PersonFunction personFunction = (PersonFunction) accountability;
+                final Function function = personFunction.getFunction();
+                if (function != null && function.getFunctionType() == FunctionType.PRESIDENT) {
+                    final Party childParty = accountability.getChildParty();
+                    if (childParty != null && childParty.isPerson()) {
+                        departmentPresidents.add((Person) childParty);
+                    }
+                }
+            }
+        }
+        return departmentPresidents;
+    }
+
+    public static boolean isPersonCurrentDepartmentPresident(Person person, Department department) {
+        return getCurrentDepartmentPresidents(department).contains(person);
```

Comment:
This implementation is very slow... why not have a method:
   public static Stream<Person> getCurrentDepartmentPresidents(Department department);
instead of the previous method? Then you can just return the first match with person or collect the entire stream but only when necessary.